hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -871,27 +879,39 @@ protected void bootstrapCommit(List<DirectoryInfo> partitionInfoList, String cre
         return HoodieMetadataPayload.createPartitionFilesRecord(
             partitionInfo.getRelativePath().isEmpty() ? NON_PARTITIONED_NAME : partitionInfo.getRelativePath(), Option.of(validFileNameToSizeMap), Option.empty());
       });
-      partitionRecords = partitionRecords.union(fileListRecords);
+      filesPartitionRecords = filesPartitionRecords.union(fileListRecords);
```

Comment:
I did leave this comment in one of the previous patches. but can we make the partition path name deduction to a method and reuse that everywhere. 
```
partitionInfo.getRelativePath().isEmpty() ? NON_PARTITIONED_NAME : partitionInfo.getRelativePath()
```
we already had some bugs around non partitioned dataset. so wanted to keep it in one place. 

also this one
```
String partition = partitionName.equals(EMPTY_PARTITION_NAME) ? NON_PARTITIONED_NAME : partitionName;
```

Reply:
Done.